### PR TITLE
Readable segment values for browser, os and country segments

### LINKED ISSUE #6772: DeviceDetection segment editor auto suggested values should show full strings
The goal of this issue is to review the Device Detection segments such as `browserCode` and `operatingSystemCode`.
- Currently the valid segment values  for `browserCode` are `FF`, `CH`, `IE`, etc.
  - would be more understandable and guessage to use `firefox`, `chrome`, `internet explorer` etc.
  - the segment auto suggest values should contain automatically all the top browsers we know of, instead of using the `Live` auto suggested feature. 
  - for Backward compatibility we should still accept `FF`, `CH` etc. and define `firefox` or `Chrome` as aliases to those ones.
- Same applies to `operatingSystemCode` where segment values would be pretty string such as `windows xp`, `windows 7`, `GNU/Linux`
  - it's case insensitive, both `Windows XP` and `windows xp` will alias to `WXP`

Implementation: refs `suggestedValuesCallback` feature.

### PR DESCRIPTION
**Pull request by sgiehl**
Current segments for browser, os or country are only usable using the short code (e.g. FF, IE, WIN, IOS, de, fr, ...).

refs #5309
fixes #6772
refs #14053

### CODE REVIEW

File: plugins/DevicesDetection/Columns/BrowserName.php [hunk 1]
```diff
@@ -17,12 +18,23 @@ class BrowserName extends Base
 {
     protected $columnName = 'config_browser_name';
     protected $columnType = 'VARCHAR(10) NULL';
-    protected $segmentName = 'browserCode';
+    protected $segmentName = 'browserName';
```
[the remark this answers is not in the document]
Comment (sgiehl) [reply plugins/DevicesDetection/Columns/BrowserName.php]: Sure. There will be two segments each `browserName` and `browserCode` and same for os.

**Comment by sgiehl** [comment]
Expected test files may need an update before merging

**Comment by sgiehl** [comment]
I've updated the branch and added an additional segment for the english country name

File: plugins/DevicesDetection/Columns/BrowserName.php [hunk 2]
```diff
@@ -17,12 +19,42 @@ class BrowserName extends Base
...
     protected $namePlural = 'DevicesDetection_Browsers';
-    protected $acceptValues = 'FF, IE, CH, SF, OP, etc.';
+    protected $acceptValues = 'FireFox, Internet Explorer, Chrome, Safari, Opera etc.';
     protected $type = self::TYPE_TEXT;
 
+    public function __construct()
+    {
+        $this->sqlFilterValue = function ($val) {
+            $browsers = Browser::getAvailableBrowsers();
+            $result   = array_search($val, $browsers);
```
Comment (diosmosis) [inline plugins/DevicesDetection/Columns/BrowserName.php]: Would it be a good idea if we made the segment case insensitive?

Comment (sgiehl) [reply plugins/DevicesDetection/Columns/BrowserName.php]: When choosing a suggest value that shouldn't matter. But shouldn't hurt to compare them lower case. Will change them

**Comment by sgiehl** [comment]
@diosmosis browser and os names should now be compared case insensitive

**Comment by diosmosis** [comment]
I'm seeing this change in the tests:

```
-    <segment>countryCode==gb</segment>
+    <segment>countryName==gb</segment>
```

Is that correct? Reporting APIs may need to be changed possibly...

**Comment by sgiehl** [comment]
@diosmosis no, I've updated the code, so that doesn't change. Tests should also be updated now.

**Pull request merged**